src/packet/secret_key.js

```diff
@@ -203,7 +203,7 @@ class SecretKeyPacket extends PublicKeyPacket {
     arr.push(new Uint8Array(optionalFieldsArr));
 
     if (!this.isDummy()) {
-      if (!this.s2kUsage) {
+      if (!this.s2kUsage || this.isStoredInHardware()) {
         this.keyMaterial = crypto.serializeParams(this.algorithm, this.privateParams);
       }
```

Comment:
I don't understand why this is needed - if the private key is stored in hardware, the private key material is not here, right? We should simply serialize the S2K specifier (which is already done earlier in the function), and nothing else, I think?

Reply:
Indeed serializing parameters was not needed here. I've corrected the condition here.